lib/views/pages/home/content/ranking/ranking_page.dart

```diff
@@ -0,0 +1,19 @@
+import "package:flutter/material.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/viewmodels/users_ranking_view_model.dart";
+import "package:proxima/views/components/async/circular_value.dart";
+import "package:proxima/views/pages/home/content/ranking/components/ranking_widget.dart";
+
+class RankingPage extends ConsumerWidget {
+  const RankingPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final ranking = ref.watch(usersRankingViewModelProvider);
+
+    return CircularValue(
+      value: ranking,
+      builder: (context, value) => RankingWidget(ranking: value),
```

Comment:
Sure, first it allows for isolated component testing of the `RankingWdiget` by simply passing mock data, also as the `RankingWdiget` is very dependent on the rankings value merging the two would mean I would have no other choice but to indent one more level for the whole build method and finally if we come to change some minor thing on the ranking page we would do it there instead of inside the ranking widget which is purely the leaderboard from a conceptual standpoint (we have a ranking page, and it happens to only contain the leaderboard).

Reply:
Yes ok, this makes a lot of sense. Thank you :)